src/WebJobs.Script/Workers/Rpc/Configuration/ProfileCondition.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc
+{
+    public enum ConditionType
+    {
+        [EnumMember(Value = "environment")]
```

Comment:
I understand the approach, but ideally, this wouldn't be an enumeration, but instead, we'd resolve the conditions from a factory and keep this extensible, so adding a condition would require no modification to existing code, just a new implementation of a condition. This would also enable conditions to be added by simply registering new "condition providers" the factory could use, which could even be done/implemented outside of the host.

Reply:
Ahh I like the sound of that!